x-pack/plugins/fleet/public/applications/fleet/components/settings_flyout/confirm_modal.tsx

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import {
+  EuiModal,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiModalFooter,
+  EuiModalBody,
+  EuiCallOut,
+  EuiButton,
+  EuiButtonEmpty,
+  EuiBasicTable,
+  EuiText,
+  EuiSpacer,
+} from '@elastic/eui';
+import type { EuiBasicTableProps } from '@elastic/eui';
+
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+
+export interface SettingsConfirmModalProps {
+  changes: Array<{
+    type: 'elasticsearch' | 'fleet_server';
+    direction: 'removed' | 'added';
+    urls: string[];
+  }>;
+  onConfirm: () => void;
+  onClose: () => void;
+}
+
+type Change = SettingsConfirmModalProps['changes'][0];
+
+const TABLE_COLUMNS: EuiBasicTableProps<Change>['columns'] = [
+  {
+    name: i18n.translate('xpack.fleet.settingsConfirmModal.fieldLabel', {
+      defaultMessage: 'Field',
+    }),
+    field: 'label',
+    render: (_, item) => getLabel(item),
+    width: '180px',
+  },
+  {
+    field: 'urls',
+    name: i18n.translate('xpack.fleet.settingsConfirmModal.valueLabel', {
+      defaultMessage: 'Value',
+    }),
+    render: (_, item) => {
+      return (
+        <EuiText size="s" color={item.direction === 'added' ? 'secondary' : 'danger'}>
+          {item.urls.map((url) => (
+            <div key={url}>{url}</div>
```

Comment:
unlikely that the user will enter the same URL twice, but it might be safer to key on index